### PR TITLE
Add tests for cli.py and some minor refactoring

### LINKED ISSUE #97: Add tests for cli.py
`cli.py` has a low test coverage (see https://codecov.io/gh/raiden-network/raiden-pathfinding-service/src/master/pathfinder/cli.py). Add more tests to increase coverage of the cli interface.

- [x] increase coverage of `cli.py`
- [x] verify that cli parameters reach the `PathfindingService` constructor

### PR DESCRIPTION
**Pull request by karlb**
Closes #97.

### CODE REVIEW

File: pathfinder/tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    PathfindingService=DEFAULT,
+    ServiceApi=DEFAULT,
+    HTTPProvider=DEFAULT,
+    get_default_registry_and_start_block=DEFAULT,
+)
+
+
+def test_bad_eth_client(caplog):
+    """ Giving a bad `eth-rpc` value should yield a concise error message
+    """
```
[the remark this answers is not in the document]
Comment (karlb) [reply pathfinder/tests/test_cli.py]: No, it's just got used to it in other projects which were doing it for consistency with multi line docstrings. I didn't find single line docstrings in PFS to compare to when writing that. I'll change it to `""" Text """`.

File: pathfinder/tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    runner = CliRunner()
+    with patch.multiple(**patch_args) as mocks:
+        mocks['get_default_registry_and_start_block'].return_value = Mock(), Mock()
+        confirmations = 77
+        result = runner.invoke(main, ['--confirmations', confirmations], catch_exceptions=False)
+        assert result.exit_code == 0
+        assert mocks['PathfindingService'].call_args[1]['required_confirmations'] == confirmations
+
+
+@pytest.mark.skip  # TODO: add a fixture provides the required contracts
```
[the remark this answers is not in the document]
Comment (karlb) [reply pathfinder/tests/test_cli.py]: I didn't find a useful fixture to get that to work, but the test would cover a useful part of the code, so I didn't discard it. Do we have a fixture that makes this work or can we easily write one? I'm still missing the details of the contract setup.

File: pathfinder/tests/test_cli.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+def test_default_registry(web3):
+    """ We can fall back to a default registry if none if specified
+    """
+    net_version = int(web3.net.version)
```
[the remark this answers is not in the document]
Comment (palango) [reply pathfinder/tests/test_cli.py]: So you need a token network registry contract?

File: pathfinder/tests/test_cli.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (palango) [reply pathfinder/tests/test_cli.py]: You need a token network registry contract?

File: pathfinder/tests/test_cli.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (karlb) [reply pathfinder/tests/test_cli.py]: Sure. I was unsure about the implications, so I kept it close to the real code.

File: pathfinder/tests/test_cli.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (karlb) [reply pathfinder/tests/test_cli.py]: Yes. When I just add the `token_network_registry_contract` fixture it fails with
```
    def get_contracts_deployed(chain_id: int, version: Optional[str] = None):
        deployment_file_path = contracts_deployed_path(chain_id, version)
    
        try:
            with deployment_file_path.open() as deployment_file:
                deployment_data = json.load(deployment_file)
        except (JSONDecodeError, UnicodeDecodeError, FileNotFoundError) as ex:
>           raise ValueError(f'Cannot load deployment data file: {ex}') from ex
E           ValueError: Cannot load deployment data file: [Errno 2] No such file or directory: '/home/karl/tools/venv-pfs/lib/python3.7/site-packages/raiden_contracts/data_pre_limits/deployment_mainnet.json'
```

File: pathfinder/tests/test_cli.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        assert result.exit_code == 0
+        assert mocks['PathfindingService'].call_args[1]['required_confirmations'] == confirmations
+
+
+def test_default_registry(token_network_registry_contract):
+    """ We can fall back to a default registry if none if specified """
+    net_version = 3
+    contracts_version = 'pre_limits'
+    registry_address, block_number = get_default_registry_and_start_block(
+            net_version,
```
Comment (palango) [inline pathfinder/tests/test_cli.py]: A bit to much indention here

File: pathfinder/tests/test_cli.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        result = runner.invoke(main, ['--confirmations', confirmations], catch_exceptions=False)
+        assert result.exit_code == 0
+        assert mocks['PathfindingService'].call_args[1]['required_confirmations'] == confirmations
+
+
+def test_default_registry(token_network_registry_contract):
+    """ We can fall back to a default registry if none if specified """
+    net_version = 3
+    contracts_version = 'pre_limits'
+    registry_address, block_number = get_default_registry_and_start_block(
```
Comment (palango) [inline pathfinder/tests/test_cli.py]: It should be checked that these are valid values and that the block number is bigger than 0

File: pathfinder/cli.py [hunk 6]
```diff
@@ -27,6 +27,27 @@
...
+def validate_address(ctx, param, value):
+    if value is None:
+        # None as default value allowed
+        return None
+    if not is_checksum_address(value):
+        raise click.BadParameter('not an EIP-55 checksummed address')
+    return value
+
+
+def get_default_registry_and_start_block(net_version, contracts_version):
```
Comment (palango) [inline pathfinder/cli.py]: types here would be nice as well

**Pull request merged**